bve-derive/src/serde_proxy.rs

```diff
@@ -191,23 +187,11 @@ fn generate_proxy_object(name: &Ident, fields: &[Field]) -> TokenStream2 {
     let new_data = fields
         .iter()
         .map(|field| match &field.ty.0 {
-            vec if vec.last().map(String::as_str) == Some("Vector1") => {
-                let original_name = &field.name;
-                let inner_type = get_first_generic_argument(vec, &field.ty.1);
-                let x_new = format_ident!("{}_x", original_name);
-                let attributes = combine_attributes(&field.attributes);
-                let (x_inner, x_conversion, x_attribute) = process_default(&x_new, &inner_type, &field.default);
-                let proxy_fields = quote! {
-                    #attributes
-                    #x_attribute
-                    #x_new: #x_inner,
-                };
-                let from_fields = quote! {
-                    #original_name: ::cgmath::Vector1::new(#x_conversion),
-                };
-                (proxy_fields, from_fields)
-            }
-            vec if vec.last().map(String::as_str) == Some("Vector2") => {
+            vec if match vec.last().map(String::as_str) {
+                Some("Vec2") | Some("BVec2") | Some("UVec2") | Some("IVec2") => true,
```

Comment:
Is matching string type names really the best way to do this?

Reply:
No, but holy fuck would doing it properly be hard. It's a hacky mess.

Reply:
I'll accept it